contrib/kafka-container-source/cmd/main.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"log"
+	"os"
+	"os/signal"
+
+	"github.com/Shopify/sarama"
+	cluster "github.com/bsm/sarama-cluster"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/client"
+	"github.com/cloudevents/sdk-go/pkg/cloudevents/types"
+	"github.com/google/uuid"
+)
+
+var (
+	sink      string
+	topic     string
+	groupID   string
+	bootstrap string
+)
+
+func init() {
+	flag.StringVar(&sink, "sink", "", "the host url to send messages to")
+	flag.StringVar(&topic, "topic", "", "the topic to read messages from")
+	flag.StringVar(&groupID, "groupId", uuid.New().String(), "the consumer group")
+	flag.StringVar(&bootstrap, "bootstrap", "localhost:9092", "Apache Kafka bootstrap servers")
+}
+
+func main() {
+
+	flag.Parse()
+
+	kafkaConfig := cluster.NewConfig()
+	kafkaConfig.Group.Mode = cluster.ConsumerModePartitions
+	kafkaConfig.Consumer.Offsets.Initial = sarama.OffsetOldest
+
+	ce, err := client.NewHTTPClient(
+		client.WithTarget(sink),
+		client.WithHTTPBinaryEncoding(),
+		client.WithUUIDs(),
+		client.WithTimeNow(),
+	)
+	if err != nil {
+		log.Fatalf("Failed to create a http cloudevent client: %s", err.Error())
+	}
+
+	// init consumer
+	brokers := []string{bootstrap}
+	topics := []string{topic}
+
+	consumer, err := cluster.NewConsumer(brokers, groupID, topics, kafkaConfig)
+	if err != nil {
+		panic(err)
+	}
+	defer consumer.Close()
+
+	// trap SIGINT to trigger a shutdown.
+	signals := make(chan os.Signal, 1)
```

Comment:
Is it better to use library function [SetupSignalHandler()](https://github.com/knative/eventing-sources/blob/0c1c5457cda74a6bdeedbce942ce878fb55e4082/vendor/github.com/knative/pkg/signals/signal.go#L29-L43)?

Reply:
@nak3 thanks, good point! I've changed that